sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -113,6 +113,11 @@ private[sql] class JSONOptions(
   }
   val lineSeparatorInWrite: String = lineSeparator.getOrElse("\n")
 
+  /**
+   * Generating JSON strings in pretty representation if the parameter enabled.
+   */
+  val pretty: Boolean = parameters.get("pretty").map(_.toBoolean).getOrElse(false)
```

Comment:
hm .. so now finally this became an actual problem. This is specifically for JSON functions and documented nowhere. Can we deal with this problem?

Reply:
Updated comments for the `json()` method of `DataFrameWriter` since we reference to it from `to_json`.